# Exact Confidence Intervals

Recall from class that if $X\sim\operatorname{Exp}(\lambda)$ for some unknown rate parameter $\lambda$, then an *exact confidence interval* for $\lambda$ with confidence level $1-\alpha$ is

$$\left(\frac{-\ln\left(1-\frac{\alpha}{2}\right)}{X}, \frac{-\ln\left(\frac{\alpha}{2}\right)}{X}\right).$$

So in particular, for a 95% confidence (i.e., $\alpha = 0.05$), we will use

$$\left(\frac{-\ln 0.975}{X}, \frac{-\ln 0.025}{X}\right).$$

Let's check this using $\lambda = 3$.  First, we'll generate a random realization of $X$ and then compute the 95% confidence interval:

In [9]:
x <- rexp(1, 3)
-log(0.975)/x
-log(0.025)/x

[1] 0.1150711

[1] 16.7662

This tells us that we probably have $\lambda\in(0.1151, 16.7662)$.  And indeed, 3 is in this interval!

In practice, we wouldn't actually know that $\lambda = 3$.  We just chose $\lambda = 3$ to both generate some random data in R and to check our work with a Monte Carlo simulation.

So now, let's check our work!  First, we'll want a way of checking whether 3 is in the confidence interval.  We have two inequalities which we can combine with **&&** (for "and"):

In [1]:
x <- rexp(1, 3); -log(0.975)/x < 3 && 3 < -log(0.025)/x

[1] TRUE

Not surprisingly, 3 is in our confidence interval.  We expect this to happen with probability 0.95.  Now let's run our simulation 10000 times and see what happens:

In [10]:
mean(replicate(10000,{x <- rexp(1, 3); -log(0.975)/x < 3 && 3 < -log(0.025)/x}))

[1] 0.9469

This is pretty close to 0.95, so our confidence interval appears to be correct!